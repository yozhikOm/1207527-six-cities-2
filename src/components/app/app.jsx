import * as React from 'react';
import PropTypes from "prop-types";
import {PureComponent} from 'react';
import {Header} from '../header/header.jsx';
import {Menu} from '../menu/menu.jsx';
import {Sorting} from '../sorting/sorting.jsx';
import {Properties} from '../properties/properties.jsx';

class App extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      offers
    } = this.props;

    return (
      <div className="page page--gray page--main">
        <Header />
        <main className="page__main page__main--index">
          <Menu />
          <div className="cities">
            <div className="cities__places-container container">
              <section className="cities__places places">
                <h2 className="visually-hidden">Places</h2>
                <b className="places__found">312 places to stay in Amsterdam</b>
                <Sorting />
                <Properties items={offers} />
              </section>
              <div className="cities__right-section">
                <section className="cities__map map"></section>
              </div>
            </div>
          </div>
        </main>
      </div>
    );
  }
}

App.propTypes = {
  offers: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        title: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
        price: PropTypes.number.isRequired,
        photos: PropTypes.arrayOf(
            PropTypes.shape({
              src: PropTypes.string,
            })
        )
      })
  )
};

export {App};
