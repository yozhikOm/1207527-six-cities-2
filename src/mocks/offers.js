export const offers = [
  {
    "id": 1,
    "title": `Cozy house amsterdam`,
    "type": `House`,
    "price": 650,
    "photos": [
      {
        "src": `img/apartment-01.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3909553943508, 4.85309666406198],
  },
  {
    "id": 2,
    "title": `Cosy room in spacious apartment`,
    "type": `Room`,
    "price": 45,
    "photos": [
      {
        "src": `img/apartment-02.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.369553943508, 4.85309666406198],
  },
  {
    "id": 3,
    "title": `Elegant private room in stylish house`,
    "type": `Room`,
    "price": 128,
    "photos": [
      {
        "src": `img/room.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3909553943508, 4.929309666406198],
  },
  {
    "id": 4,
    "title": `Huge Tinyhouse in the green, between sea and city`,
    "type": `House`,
    "price": 1520,
    "photos": [
      {
        "src": `img/apartment-03.jpg`
      },
      {
        "src": `img/amsterdam.jpg`
      },
    ],
    "coordinates": [52.3809553943508, 4.939309666406198],
  }
];
