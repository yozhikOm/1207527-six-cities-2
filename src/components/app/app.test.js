import React from 'react';
import {App} from './app.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';
import {offers} from '../../mocks/offers.js';

it(`App компонент рендерится корректно`, () => {
  const appProps = {
    mainPageProps: {
      offers: {offers}.offers
    },
    propertyCardProps: {
      offer: {offers}.offers[0]
    }
  };

  const renderer = new ShallowRenderer();
  renderer.render(<App {...appProps} />);

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();

  // const appComponent = renderer.create(<App {...appProps} />).toJSON();
  // expect(appComponent).toMatchSnapshot();
});
