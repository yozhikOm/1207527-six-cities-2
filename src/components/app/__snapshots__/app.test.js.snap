// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App компонент рендерится корректно 1`] = `
<React.Fragment>
  <MainPage
    offers={
      Array [
        Object {
          "coordinates": Array [
            52.3909553943508,
            4.85309666406198,
          ],
          "id": 1,
          "photos": Array [
            Object {
              "src": "img/apartment-01.jpg",
            },
            Object {
              "src": "img/amsterdam.jpg",
            },
          ],
          "price": 650,
          "title": "Cozy house amsterdam",
          "type": "House",
        },
        Object {
          "coordinates": Array [
            52.369553943508,
            4.85309666406198,
          ],
          "id": 2,
          "photos": Array [
            Object {
              "src": "img/apartment-02.jpg",
            },
            Object {
              "src": "img/amsterdam.jpg",
            },
          ],
          "price": 45,
          "title": "Cosy room in spacious apartment",
          "type": "Room",
        },
        Object {
          "coordinates": Array [
            52.3909553943508,
            4.929309666406198,
          ],
          "id": 3,
          "photos": Array [
            Object {
              "src": "img/room.jpg",
            },
            Object {
              "src": "img/amsterdam.jpg",
            },
          ],
          "price": 128,
          "title": "Elegant private room in stylish house",
          "type": "Room",
        },
        Object {
          "coordinates": Array [
            52.3809553943508,
            4.939309666406198,
          ],
          "id": 4,
          "photos": Array [
            Object {
              "src": "img/apartment-03.jpg",
            },
            Object {
              "src": "img/amsterdam.jpg",
            },
          ],
          "price": 1520,
          "title": "Huge Tinyhouse in the green, between sea and city",
          "type": "House",
        },
      ]
    }
  />
</React.Fragment>
`;
